### PR TITLE
Add connect commands + selector

### PR DESCRIPTION
**Pull request by queer**
- Builds commands for each plugin from project env
- Provides menu for selecting plugin to connect to

### CODE REVIEW

File: cmd/connect.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+	fmt.Printf("🎉 Connecting to: %s %s\n", ui.MagentaText(project.Name), ui.MagentaText(envName))
+
+	names := make([]string, 0)
+	for _, plugin := range project.Plugins {
+		// TODO: Better way of handling this
+		if plugin.Name != "env" {
+			names = append(names, plugin.Name)
+		}
+	}
```
Comment (queer) [inline cmd/connect.go:31]: This is a bit of an ugly thing here, but I just didn't see a better way to handle this...

Comment (JakeCooper) [reply cmd/connect.go:31]: @coffee-cup Another env oopsie :<

Wonder if there's a way to move this before it becomes a "founding principle"

Comment (JakeCooper) [reply cmd/connect.go:31]: e.g we ship binaries with logic for it

Comment (coffee-cup) [reply cmd/connect.go:31]: We should probably have a `visiblePlugins` function that filters out env plugins. I can see at one point we would have multiple env plugins.

**Review by JakeCooper — APPROVED** [review]
I think this is dope, but I would also want a "hot" command that I could just type to connect. Thoughts?

`railway psql` or `railway connect psql` or `railway run psql`

I think the latter COULD be good, because it would seem as thought we're truly running that command "in the environment". My only thoughts is kinda overloading the "run" command

File: cmd/connect.go (hunk 1 above)
Comment (coffee-cup) [reply cmd/connect.go:31]: could also just stop sending back "env" as part of the project plugins. But I imagine that would break a few other things

**Comment by coffee-cup** [comment]
Should Wondering if we should have some sort of help message if the connect command fails? Something like "Connecting failed? Do you have psql installed?"

File: cmd/connect.go (hunk 1 above)
Comment (coffee-cup) [reply cmd/connect.go:31]: Actually, in this case we might want a `getDatabasePlugins` since we will (hopefully) have other non-db plugins in the future

**Comment by JakeCooper** [comment]
Yea maybe a little error message. Obvs stdout will have it but still might be nice.

**Comment by queer** [comment]
> I think this is dope, but I would also want a "hot" command that I could just type to connect. Thoughts?

:+1: Added.

**Pull request merged**